kube-runtime/src/controller/mod.rs

```diff
@@ -234,6 +234,7 @@ impl Display for ReconcileReason {
 }
 
 const APPLIER_REQUEUE_BUF_SIZE: usize = 100;
+const SCHEDULER_DEBOUNCE_PERIOD: Duration = Duration::from_secs(1);
```

Comment:
Thinking more about the danger element, maybe we should just leave this at zero for instant response (which is probably more attractive while developing).

That way we can keep it in the `impl Default for Config`.

We can also pass `controller::Config` to `applier` directly (which makes it easy and more transparent to see in tests - where you are currently passing an opaque `None` - as an extra arg `Config::default()). This way we also don't have to unwrap_or inside the applier. We should always pass a debounce into scheduler (but usually it will be zero).

Reply:
> We should always pass a debounce into scheduler

why exactly? i agree 100% that we should pass `controller::Config` directly to `applier` but i think we should be able to pass `None` as a debounce to the `kube_runtime::scheduler()`. since it is a public API, i feel it'll be easier for users to transition, since they can just pass `None`. about avoiding `.unwrap_or(PERIOD)` inside `applier`, we can just pass `Config.debounce` directly since the PR already does `debounce.unwrap_or(Duration::ZERO)` in `kube_runtime::scheduler()` and the now decided default debounce period is zero seconds.

Reply:
> i feel it'll be easier for users to transition, since they can just pass None

yeah, i agree with that and see your concern. there is also a competing goal; keeping our code more readable, and opaque nulls in test code inflate the tests (new line everywhere) and make things non-obvious (outside of an editor with inlay hints that shows what the null means). one way to satisfy both goals (and avoid people having to import durations for them to only set to zero) would be:

- create a `schedule_debounced` ctor that calls `Scheduler::new` with an explicit duration (non-optional)
- leave `fn scheduler` in place (only allowing zero debounce - making scheduler change non-breaking)
- make `applier` use `schedule_debounced` (since applier always has a number)
- make tests either use `scheduler` or `schedule_debounced` depending on whether or not they are testing debounce behavior